crates/namada/src/ledger/native_vp/masp.rs

```diff
@@ -337,9 +403,227 @@ where
+                // to this event
+                let address =
+                    MaybeIbcAddress::Ibc(packet_data.receiver.to_string());
+                // Public keys must be the hash of the sources/targets
+                let address_hash = TransparentAddress(<[u8; 20]>::from(
+                    ripemd::Ripemd160::digest(sha2::Sha256::digest(
+                        &address.serialize_to_vec(),
+                    )),
+                ));
+                changed_balances.decoder.insert(address_hash, address);
+                changed_balances
+                    .pre
+                    .entry(address_hash)
+                    .or_insert(ValueSum::zero());
+                changed_balances
+                    .post
+                    .entry(ibc_address_hash)
+                    .or_insert(ValueSum::zero());
+                // Required for the IBC internal Address to release funds
+                let pre_entry = changed_balances
+                    .pre
+                    .entry(ibc_address_hash)
+                    .or_insert(ValueSum::zero());
+                // Required for the packet's receiver to get funds
+                let post_entry = changed_balances
+                    .post
+                    .entry(address_hash)
+                    .or_insert(ValueSum::zero());
+                
+                match ibc_event.kind().sub_domain() {
+                    // This event is emitted on the sender
+                    SEND_PACKET_EVENT => {
+                        // Ensure that the event corresponds to the current changes to
+                        // storage
+                        let commitment_key = storage::commitment_key(
+                            &msg.port_id_on_a,
+                            &msg.chan_id_on_a,
+                            msg.seq_on_a,
+                        );
+                        if !keys_changed.contains(&commitment_key) {
+                            // Otherwise ignore this event
+                            continue;
+                        }
+                        let Some(storage_commitment): Option<PacketCommitment> = self
+                            .ctx
+                            .read_bytes_post(&commitment_key)?
+                            .map(Into::into) else {
+                                // Ignore this event if it does not exist
+                                continue;
+                            };
+                        // Check that the packet commitment in storage is the same as
+                        // that derived from this event
+                        let packet_commitment = VpValidationContext::<'a, 'a, S, CA>::compute_packet_commitment(
+                            &msg.data,
+                            &msg.timeout_height_on_b,
+                            &msg.timeout_timestamp_on_b,
+                        );
+                        if packet_commitment != storage_commitment {
```

Comment:
Understood. I'm working on a variant of this PR where the timeout values are extracted from tx_data. I'm just a little concerned that this will cause the VP to be dependent on the transaction format instead of the state changes. Also, I haven't yet checked how this will interact with the removal of `Transfer` dependence changes in #3232 .